File: 2017/Animators/SimpleViewPropertyAnimator/app/src/main/java/me/mehdi/simpleviewpropertyanimator/MainActivity.java
package me.mehdi.simpleviewpropertyanimator;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    Button startButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        startButton = findViewById(R.id.start);

        startButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                view.animate()
                        .translationY(300)
                        .scaleX(2)
                        .alpha(0.5f)
                        .setDuration(2000)
                        .withEndAction(new Runnable() {
                            @Override
                            public void run() {
                                Toast.makeText(MainActivity.this, "The Animation Ended", Toast.LENGTH_LONG).show();
                            }
                        });
            }
        });
    }
}

File: 2022/Animators/Project/app/src/main/java/com/example/animators/MainActivity.kt
package com.example.animators

import android.animation.AnimatorInflater
import android.animation.AnimatorSet
import android.animation.ObjectAnimator
import android.animation.ValueAnimator
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Toast
import com.example.animators.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)


        binding.hello.setOnClickListener {button->
//            val xScale = ObjectAnimator.ofFloat(button, View.SCALE_X, 5f)
//
//            xScale.repeatCount = 3
//            xScale.repeatMode = ValueAnimator.REVERSE
//            xScale.duration = 1000
//
//            val yScale = ObjectAnimator.ofFloat(button, View.SCALE_Y, 4f)
//            yScale.repeatCount = 3
//            yScale.repeatMode = ValueAnimator.REVERSE
//            yScale.duration = 1000
//
////            val rotate = ObjectAnimator.ofFloat(binding.loudspeaker, View.ROTATION, 1000f)
//
//            val animatorSet = AnimatorSet()
//
//            animatorSet.playTogether(xScale, yScale)
//
//            animatorSet.start()
//
//            binding.loudspeaker.animate()
//                .rotation(1000f)
//                .xBy(200f)
//                .yBy(100f)
//                .alpha(0.5f)
//                .withEndAction {
//                    Toast.makeText(this, "Animation Finished",
//                    Toast.LENGTH_LONG).show()
//                }
//                .start()

            val resAnimator = AnimatorInflater.loadAnimator(
                this, R.animator.my_animator
            )

            resAnimator.setTarget(binding.loudspeaker)

            resAnimator.start()






        }
    }
}